### PR TITLE
VSF2-146 - VSF-121 - Disabled buttons if product is out of stock, stock display, improved quantity selector logic + variant selector with changes

### PR DESCRIPTION
**Pull request by DawidFrankiewicz**

### CODE REVIEW

**Review by dead-playmod — CHANGES_REQUESTED** [review]

File: packages/theme/pages/Product.vue [hunk 1]
```diff
@@ -32,10 +32,30 @@
...
-          <SfPrice
-            :regular="$n(productGetters.getPrice(product).regular, 'currency')"
-            :special="productGetters.getPrice(product).special && $n(productGetters.getPrice(product).special, 'currency')"
-          />
+          <div class="product__price-and-stock">
+            <SfPrice
+              :regular="$n(productGetters.getPrice(product).regular, 'currency')"
+              :special="productGetters.getPrice(product).special && $n(productGetters.getPrice(product).special, 'currency')"
+            />
+            <div v-if="product.selectedVariant.tracked" :class="`stock-info ${productGetters.isInStock(product.selectedVariant) ? '' : 'danger'}`">
```
[the remark this answers is not in the document]
Comment (DawidFrankiewicz) [reply packages/theme/pages/Product.vue:40]: Changed according to coment

File: packages/theme/components/VariantSelector.vue [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+      class="modal__add-to-cart"
+      @click="handleAddToCart"
+    >
+      <template #quantity-select-input>
+        <QuantitySelector
+          v-model="qty"
+          @input="qty = $event"
+          :qty="1"
+          :min="1"
+          :max="product.selectedVariant.tracked ? productGetters.getStockForVariant(product.selectedVariant) : 999"
```
[the remark this answers is not in the document]
Comment (DawidFrankiewicz) [reply packages/theme/components/VariantSelector.vue:77]: done

File: packages/theme/pages/Product.vue [hunk 3]
```diff
@@ -80,14 +100,28 @@
...
+              class="product__add-to-cart"
+              @click="handleAddToCart({ product, quantity: parseInt(qty) })"
+            >
+              <template #quantity-select-input>
+                <QuantitySelector
+                  v-model="qty"
+                  @input="qty = $event"
+                  :qty="1"
+                  :min="1"
+                  :max="product.selectedVariant.tracked ? productGetters.getStockForVariant(product.selectedVariant) : 999"
```
[the remark this answers is not in the document]
Comment (DawidFrankiewicz) [reply packages/theme/pages/Product.vue:118]: done

File: packages/composables/src/getters/productGetters.ts [hunk 4]
```diff
@@ -124,6 +124,21 @@ export const getProductOptions = (product: Product): any => {
   return options;
 };
 
+export const getProductStockForVariant = (variant): number => {
+  return variant ? (variant as any).onHand - (variant as any).onHold : 0;
+};
+
+const getProductQuantityLimit = (variant) => {
+  if (!variant.tracked) {
```
[the remark this answers is not in the document]
Comment (DawidFrankiewicz) [reply packages/composables/src/getters/productGetters.ts]: done

**Review by dead-playmod — APPROVED** [review]

**Pull request merged**